### PR TITLE
added new snippets, syntax and build-system

### PR DESCRIPTION
**Pull request by imranismail**
## New Snippets
- `gene` - GenEvent with callbacks
- `gens` - GenServer with callbacks
- `sup` - Supervisor with callbacks
- `app` - Application snippet
- `do` - Fixes `do` block not closing with an `end`
- `cond` - Self explanatory
- `case` - Self explanatory
- `if` - Self explanatory
- `test` - Self explanatory
## Additional Syntax Highlighting
- `test`
- `setup`
- `assert`
- `assert_receive`
- `assert_received`
- `assert_in_delta`
- `assert_raise`
- `reflute`
- `reflute_receive`
- `reflute_received`
- `reflute_in_delta`
- `flunk`
- `catch_error`
- `catch_exit`
- `catch_throw`
## Faster Build System

No longer relying on shell (bash/zsh) to run build commands (no need to wait for the shell to load)

### CODE REVIEW

File: Elixir Build.sublime-build [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "cmd": ["elixir", "${file}"],
+  "selector": "source.elixir",
+  "osx":{
+    "path": "/usr/local/bin/:${PATH}"
```
Comment (josevalim) [inline Elixir Build.sublime-build]: Will this only work if Elixir is in the path above?

Comment (imranismail) [inline Elixir Build.sublime-build]: This will only work if installed using the official guide.

Comment (josevalim) [inline Elixir Build.sublime-build]: Yes, so I am :-1:. A lot of folks use things like kiex or exenv. I personally use a custom directory and so on. :)

Comment (imranismail) [inline Elixir Build.sublime-build]: I'll revert to the old ones then! :+1:

Comment (imranismail) [inline Elixir Build.sublime-build]: done

**Comment by josevalim** [comment]
Thanks @imranismail. can we please not add the supervisor, genevent and genserver ones in this PR? I am not convinced they are a good addition because I am asking myself tons of questions about what should be the defaults (like moduledoc false), which options to pass, how to name variables. Everything else looks great, thank you!

**Comment by endersstocker** [comment]
@imranismail Behaviour macros (DSL keywords) should not be identified as Elixir keywords in the syntax. They are not reserved keywords; a programmer can name a variable or function `test` or `setup` for instance. Secondly, it is not possible to color all possible behaviour macros. It would be confusing for `ExUnit.Case` to get special treatment.

I agree with @josevalim about GenEvent and GenServer. The other snippets look great.

**Comment by imranismail** [comment]
Thanks for the feedback, I will update the changes accordingly.

**Comment by josevalim** [comment]
Closing as there has been no changes, thank you.

**Pull request closed without merging**